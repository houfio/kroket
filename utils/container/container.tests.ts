import { renderHook } from '@testing-library/react-hooks';

import { useContainer } from '.';

it('should create a container if it doesn\'t exist', () => {
  const { result: { current } } = renderHook(() => useContainer('test'));

  expect(document.body).toMatchSnapshot();
  expect(current).toMatchSnapshot();
});

it('should return the container if it exists', () => {
  document.body.innerHTML = `
    <div id="test"></div>
  `;

  const { result: { current } } = renderHook(() => useContainer('test'));

  expect(document.body).toMatchSnapshot();
  expect(current).toMatchSnapshot();
});
