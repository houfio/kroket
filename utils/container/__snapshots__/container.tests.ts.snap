// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a container if it doesn't exist 1`] = `
<body>
  <div
    id="test"
  />
</body>
`;

exports[`should create a container if it doesn't exist 2`] = `
<div
  id="test"
/>
`;

exports[`should return the container if it exists 1`] = `
<body>
  
    
  <div
    id="test"
  />
  
  
</body>
`;

exports[`should return the container if it exists 2`] = `
<div
  id="test"
/>
`;
